client/src/redux/index.js

```diff
@@ -655,7 +667,9 @@ export const reducer = handleActions(
                 ...state.user[appUsername].completedChallenges
               ],
               'id'
-            )
+            ),
+            savedChallenges:
+              savedChallenges ?? savedChallengesSelector(state[MainApp])
```

Comment:
Since `submitComplete` is generic, it might be a good idea to only send the savedChallenges to the client when a specific challenge has been submitted?

Reply:
Since we send the completedChallenges back and forth between the api and the client, optimizing this route might not   be necessary.